paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsAdapter.kt

```diff
@@ -428,6 +448,145 @@ internal class PaymentOptionsAdapter(
+                R.string.stripe_paymentsheet_remove_pm,
+                getDescription(resources)
+            )
+        }
+    }
+    internal companion object {
+        private fun calculateViewWidth(parent: ViewGroup): Dp {
+            val targetWidth = parent.measuredWidth - parent.paddingStart - parent.paddingEnd
+            val screenDensity = parent.context.resources.displayMetrics.density
+            // minimum width for each item, accounting for the CardView margin so that the CardView
+            // is at least 100dp wide
+            val minItemWidth = 100 * screenDensity + 6.0f + 6.0f
+            // numVisibleItems is incremented in steps of 0.5 items (1, 1.5, 2, 2.5, 3, ...)
+            val numVisibleItems = (targetWidth * 2 / minItemWidth).toInt() / 2f
+            val viewWidth = (targetWidth / numVisibleItems)
+            return (viewWidth / screenDensity).dp
+        }
+    }
+}
+
+@Composable
+fun PaymentOptionUi(
+    viewWidth: Dp,
+    isSelected: Boolean,
+    isEditing: Boolean,
+    isEnabled: Boolean,
+    iconRes: Int,
+    accessibilityDescription: String,
+    labelText: String? = "",
+    onRemoveListener: (() -> Unit)? = null,
+    onRemoveAccessibilityDescription: String = "",
+    onItemSelectedListener: (() -> Unit)
+) {
+    val strokeColor = colorResource(
+        if (isSelected) {
+            R.color.stripe_paymentsheet_payment_option_selected_stroke
+        } else {
+            R.color.stripe_paymentsheet_payment_option_unselected_stroke
+        }
+    )
+
+    val cardBackgroundColor = if (isEnabled) {
+        MaterialTheme.colors.surface
+    } else {
+        colorResource(id = R.color.stripe_paymentsheet_elements_background_disabled)
+    }
+
+    ConstraintLayout(
+        modifier = Modifier
+            .padding(top = 12.dp)
+            .width(viewWidth)
+            .selectable(selected = isSelected, enabled = isEnabled, onClick = {
+                onItemSelectedListener()
+            })
+    ) {
+        val (checkIcon, deleteIcon, label, card) = createRefs()
+        Card(
+            border = BorderStroke(if (isSelected) 2.dp else 1.dp, strokeColor),
+            shape = RoundedCornerShape(6.dp),
+            elevation = 2.dp,
+            backgroundColor = cardBackgroundColor,
+            modifier = Modifier
+                .height(64.dp)
+                .padding(start = 6.dp, end = 6.dp)
+                .fillMaxWidth()
+                .constrainAs(card) {
+                    top.linkTo(parent.top)
+                    start.linkTo(parent.start)
+                    end.linkTo(parent.end)
+                }
+        ) {
+            Column(
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally,
+            ) {
+                Image(
+                    alpha = if (isEnabled) 1.0F else 0.6F,
+                    painter = painterResource(iconRes),
+                    contentDescription = "",
```

Comment:
Just double checking that this isn't needed for accessiblity

Reply:
same as above. The empty string prevents the reader from mentioning the image.

Reply:
Nit: It should be null instead of empty.

> Some visual elements are purely decorative and you might not want to communicate them to the user. When you set the contentDescription parameter to null, you indicate to the Android framework that this element does not have associated actions or state.

https://developer.android.com/jetpack/compose/accessibility